compile failure with gcc 4.8.5 (RedHat 7.3)

From email report:

I experienced compilation problem in libfabric using Intel compilers (18.0.1.163 Build 20171018) interfaced to the default gcc 4.8.5 20150623 (Red Hat 4.8.5-11) in the "Red Hat Enterprise Linux Server release 7.3 (Maipo)" platform. Switching to load gcc 6.3.0 module in the same platform could, however, compile successfully. Jeff Hammond suggested me to post the problem here, and I attach the compilation error messages for your reference.

wget https://github.com/ofiwg/libfabric/archive/master.zip
unzip master.zip
mv libfabric-master libfabric
cd libfabric

./autogen.sh 
../configure CC=icc CXX=icpc --enable-psm2 --disable-udp --disable-sockets --disable-rxm --prefix=$NWCHEM_ROOT/deps

make

make  all-am
make[1]: Entering directory `/work1/home/u1/molpro/src/nwchem-6.8.1.opa.x/tests/libfabric/build'
  CC       src/src_libfabric_la-fabric.lo
In file included from ../include/ofi.h(50),
                 from ../include/ofi_util.h(55),
                 from ../src/fabric.c(44):
../include/ofi_atom.h(66): error: identifier "atomic_int_least32_t" is undefined
        typedef atomic_int_least32_t    ofi_atomic_int32_t;
                ^

In file included from ../include/ofi.h(50),
                 from ../include/ofi_util.h(55),
                 from ../src/fabric.c(44):
../include/ofi_atom.h(67): error: identifier "atomic_int_least64_t" is undefined
        typedef atomic_int_least64_t    ofi_atomic_int64_t;
                ^

compilation aborted for ../src/fabric.c (code 2)
make[1]: *** [src/src_libfabric_la-fabric.lo] Error 1
make[1]: Leaving directory `/work1/home/u1/molpro/src/nwchem-6.8.1.opa.x/tests/libfabric/build'
make: *** [all] Error 2

--- a/configure.ac
+++ b/configure.ac
@@ -208,6 +208,21 @@ AC_TRY_LINK([#include <stdatomic.h>],
     ],
     [AC_MSG_RESULT(no)])
 
+
+AC_MSG_CHECKING(compiler support for c11 atomic `least` types)
+AC_TRY_LINK([#include <stdatomic.h>],
+    [atomic_int_least32_t a;
+     atomic_int_least64_t b;
+    ],
+    [
+        AC_MSG_RESULT(yes)
+        AC_DEFINE(HAVE_ATOMICS_LEAST_TYPES, 1,
+                  [Set to 1 to use c11 atomic `least` types])
+    ],
+    [
+        AC_MSG_RESULT(no)
+    ])
+
 dnl Check for gcc built-in atomics
 AC_MSG_CHECKING(compiler support for built-in atomics)
 AC_TRY_LINK([#include <stdint.h>],

--- a/include/ofi_atom.h
+++ b/include/ofi_atom.h
@@ -63,8 +63,13 @@ extern "C" {
 #endif
 
 #ifdef HAVE_ATOMICS
-	typedef atomic_int_least32_t	ofi_atomic_int32_t;
-	typedef atomic_int_least64_t	ofi_atomic_int64_t;
+#ifdef HAVE_ATOMICS_LEAST_TYPES
+typedef atomic_int_least32_t	ofi_atomic_int32_t;
+typedef atomic_int_least64_t	ofi_atomic_int64_t;
+#else
+typedef atomic_int	ofi_atomic_int32_t;
+typedef atomic_long	ofi_atomic_int64_t;
+#endif
 
 #define OFI_ATOMIC_DEFINE(radix)									\
 	typedef struct {										\

